src/gt4py/cartesian/gtscript.py

```diff
@@ -274,6 +286,8 @@ def stencil(
         raise ValueError(f"Invalid 'raise_if_not_cached' bool value ('{raise_if_not_cached}')")
     if cache_settings is not None and not isinstance(cache_settings, dict):
         raise ValueError(f"Invalid 'cache_settings' dictionary ('{cache_settings}')")
+    if not isinstance(literal_precision, int) and literal_precision not in (32, 64):
+        raise ValueError(f"Invalid 'literal_precision' ('{literal_precision}')")
```

Comment:
This check would be redundant with the proposed custom type descriptor for literals.